mixins/WidgetMixin.js

```diff
@@ -184,35 +199,42 @@ module.exports = {
 
     // @todo image delete_sign / checkmark should be editable via option
     // @todo options enable live validation
-    if (!(typeof this.state.value === 'undefined' || this.state.value === '') && this.state.validationErrorMessage !== null && this.props.image !== null && this.props.type !== 'OptionWidget' && this.props.validationImage === true && toValidate === true) {
+
+    let hasValue = typeof this.state.value !== 'undefined' && this.state.value !== '';
+
+    if (this.props.validateOnEmpty) {
+      hasValue = true;
+    }
+
+    const hasValidationErrors = this.state.validationErrorMessage !== null;
+    const hasImageProp = this.props.image !== null;
+    const isOptionWidget = this.props.type === 'OptionWidget'
+    const shouldShowValidationImage = this.props.validationImage === true;
+
+    if (hasValue && hasImageProp && !isOptionWidget && shouldShowValidationImage && toValidate) {
+      const imageSrc = hasValidationErrors ? 'delete_sign.png' : 'checkmark.png';
+
       return (
         <Image
           style={this.getStyle('rowImage')}
           resizeMode={Image.resizeMode.contain}
-          source={require('../icons/delete_sign.png')}
+          source={require(`../icons/${imageSrc}`)}
```

Comment:
@FaridSafi I'm using this branch in my own project and it's working perfectly. If there are no other issues with this PR, I would appreciate if you could review it again and possibly merge it into master. Thanks!